cpp/src/parquet/column_reader.cc

```diff
@@ -70,6 +70,8 @@ namespace {
 // The minimum number of repetition/definition levels to decode at a time, for
 // better vectorized performance when doing many smaller record reads
 constexpr int64_t kMinLevelBatchSize = 1024;
+// The max buffer size of validility bitmap for skipping buffered levels.
+constexpr int64_t kMaxSkipLevelBufferSize = 128;
```

Comment:
Did you get a chance to benchmark different value for kMaxSkipLevelBufferSize?

Reply:
tried 1024 byte size and got similar results. It basically correlates with the number of  levels being buffered. I tend to keep it small to avoid occupying too much memory as it's per column buffer.

Reply:
Other part is LGTM, however, when batchSize is always greater than 1024, this optimization would be a waste of memory?

Reply:
yeah, for large batch the buffer is always expanded-then-shrinked. Performance-wise it's probably is fine (as shown in the 1/5000/1000000 test where the batch size is 5k) as the original behavior is allocate-then-free. And if large batch happens there would be fewer rounds of reads/skips (e.g., for an 1M-row group it won't have more than 1K shrinks when batch size is 1025). So even if single skip incurs some overhead on the "shrink" part, the overall impact should be small.  Memory-wise that's why i set it as 128 to avoid wasting too much memory.  Alternatively we can set a global budget evenly shared by columns.

Reply:
Reallocating could be more costly than simply deallocating, since you need to copy the data when reallocating. I'm not saying this is relevant here, just pointing out that reallocation is mostly better _if you need to keep the actual data, not just the memory area_.

Reply:
yeah, i think memcpy 128B is probably fine here (typically two/three cache lines), comparing to reallocation